inc/lift/Request.h

```diff
@@ -349,6 +433,45 @@ class Request {
     /// libuv will call this function when the AddRequest() function is called.
     friend auto requests_accept_async(
         uv_async_t* handle) -> void;
+    
+    friend auto on_response_wait_time_expired_callback(uv_timer_t* handle) -> void;
+};
+
+/**
+ * Class wrapping information used by EventLoop to "time out" requests that have not
+ * received a response within the expected wait time.
+ */
+class ResponseWaitTimeWrapper
+{
+private:
+    struct Data
+    {
+        /**
+         * Represents the time point when the corresponding request should be timed out.
+         * Using uint64_t for consistency since uv_now returns time points as uint64_t.
+         */
+        uint64_t m_timeout_time;
+        /// Reference to the Request associated with this wrapper.
+        std::shared_ptr<SharedRequest>* m_shared_request_ptr_pointer;
```

Comment:
Pointer to shared_ptr seems odd.
Why not just copy the shared_ptr or use an actual reference (assuming thread-safety not required)?

Reply:
It is weird, but I think it better encapsulates the fact that we are dealing with data on the heap, which we want to keep around separate from this class. If we used a reference, we'd later have to dereference it to get the pointer, so using a raw pointer makes more sense here to me.

Reply:
By ref I meant `std::shared_ptr<SharedRequest>&` -- though this is not safe for multiple threads to use (without mutex).

Why not just use `shared_ptr<...>` instead of `unique_ptr<shared_ptr<...>>`?
Shouldn't need 2 layers here.

Reply:
Yeah, this should be a regular shared pointer.